libyul/CompilabilityChecker.cpp

```diff
@@ -48,7 +48,7 @@ map<YulString, int> CompilabilityChecker::run(
 
 		BuiltinContext builtinContext;
 		builtinContext.currentObject = &_object;
-		for (auto name: _object.dataNames())
+		for (auto name: _object.qualifiedDataNames())
```

Comment:
This puts qualified names in `BuiltinContext` while `EVMObjectCompiler` puts unqualified ones (i.e. only one level). Won't this cause some kind of a problem?

Maybe it would be possible to create a test case covering this?

Reply:
What about this one?

Reply:
What about aligning `CompilabilityChecker` with `EVMObjectCompiler::compile`?

Reply:
Honestly, I don't know much about `CompilabilityChecker`. It looks like it also creates assemblies but does it in a slightly different way that seems to depend on how `dataNames()` worked (no nesting?). I think we need another opinion here. Maybe @chriseth could take a look?

Reply:
It should work in exactly the same way as the "real" compiler just that it does not actually store the code and thus the hope is that it is faster.

Reply:
I think it is now `CompilabilityChecker` does the same thing in terms of subIds as "real" compiler.